### PR TITLE
Add ui/metrics feature flags

### PR DESCRIPTION
**Pull request by dae**
Partially implements #729:

- ~The burn crate no longer enables the training module unless the train feature is enabled.~
- Added new flags so the UI and/or system metrics can be excluded.

textplots has two security warnings that can be avoided when the CLI
is excluded:

33 │ atty 0.2.14 registry+https://github.com/rust-lang/crates.io-index
   │ ----------------------------------------------------------------- unsound advisory detected
   │
   = ID: RUSTSEC-2021-0145
   = Advisory: https://rustsec.org/advisories/RUSTSEC-2021-0145
   = On windows, `atty` dereferences a potentially unaligned pointer.

16 │ ansi_term 0.12.1 registry+https://github.com/rust-lang/crates.io-index
   │ ---------------------------------------------------------------------- unmaintained advisory detected
   │
   = ID: RUSTSEC-2021-0139
   = Advisory: https://rustsec.org/advisories/RUSTSEC-2021-0139
   = The maintainer has advised that this crate is deprecated and will not receive any maintenance.

### Checklist

- [x] Confirm that `run-checks` script has been executed.

### Related Issues/PRs

#729

### CODE REVIEW

File: burn-train/Cargo.toml [hunk 1]
```diff
@@ -10,21 +10,31 @@ readme = "README.md"
...
 
+[features]
+cli = [
+    "indicatif",
+    "rgb",
+    "terminal_size",
+    "textplots",
+    "nvml-wrapper",
+    "sysinfo",
+    "systemstat"
```
Comment (nathanielsimard) [inline burn-train/Cargo.toml:21]: Those are used to compute metrics as well, you may still want them to be computed without the CLI to show them with a plot.

Probably not `nvml-wrapper`, since it's just text, but the other two can be useful regardless.

File: burn-train/src/metric/dashboard/plot.rs [hunk 2]
```diff
@@ -146,4 +147,14 @@ impl TextPlot {
...
     }
+
+    /// Renders the text plot.
+    ///
+    /// # Returns
+    ///
+    /// The rendered text plot.
+    #[cfg(not(feature = "cli"))]
+    pub fn render(&self) -> String {
+        panic!("metrics feature not enabled on burn-train")
```
Comment (nathanielsimard) [inline burn-train/src/metric/dashboard/plot.rs:158]: I think it should be the `cli` feature, there is no `metric` feature.

File: burn-train/src/metric/mod.rs [hunk 3]
```diff
@@ -6,20 +6,30 @@ pub mod state;
...
+#[cfg(feature = "cli")]
 pub use cpu_use::*;
+#[cfg(feature = "cli")]
 pub use cuda::*;
+#[cfg(feature = "cli")]
 pub use gpu_temp::*;
 pub use learning_rate::*;
 pub use loss::*;
+#[cfg(feature = "cli")]
 pub use memory_use::*;
```
Comment (nathanielsimard) [inline burn-train/src/metric/mod.rs:35]: Except for the `cuda` metric, they should be enabled even without the cli feature.

File: burn/Cargo.toml [hunk 4]
```diff
@@ -11,12 +11,14 @@ repository = "https://github.com/burn-rs/burn"
...
-default = ["std", "train"]
+default = ["std"]
 experimental-named-tensor = ["burn-core/experimental-named-tensor"]
 std = [
   "burn-core/std",
 ]
-train = ["std", "burn-train"] # Training requires std
+# Training requires std
+train = ["std", "burn-train"]
+train-cli = ["train", "burn-train/cli"]
```
Comment (nathanielsimard) [inline burn/Cargo.toml:21]: I think `train-cli` should be enabled by default without having to specify `cli`. It is the most common usecase.

**Review by nathanielsimard — CHANGES_REQUESTED** [review]
See my comments bellow. Having a cli feature flag is fine, but I would enable it by default. Also we should have a better implementation without using text-plot, probably using something like ratatui: https://github.com/ratatui-org/ratatui.

File: burn/Cargo.toml (hunk 4 above)
Comment (antimora) [reply burn/Cargo.toml:21]: Can we also document (inline is fine) that explains the difference between train and train-cli?

File: burn-train/Cargo.toml (hunk 1 above)
Comment (dae) [reply burn-train/Cargo.toml]: Ok, I've split these off into a separate flag - does that work?

**Review by nathanielsimard — APPROVED** [review]
Thanks @dae I'm still unsure about the feature flag names `metrics` and `ui`, but at least we offer the option now!

**Comment by dae** [comment]
Conflicts fixed . No preferences on names - some other ideas: train-tui, train-sys-metrics.

**Review by nathanielsimard — APPROVED** [review]
